### PR TITLE
Use runTests in fuchsia tester.

### PR DESCRIPTION
**Pull request by garymm**
This allows us to collect code coverage.

### CODE REVIEW

**Comment by ricelid** [comment]
@cbracken @tvolkert please take a look.

**Review by cbracken — APPROVED** [review]

**Comment by cbracken** [comment]
lgtm

File: packages/flutter_tools/bin/fuchsia_tester.dart [hunk 1]
```diff
@@ -70,16 +81,37 @@ Future<Null> run(List<String> args) async {
...
+    if (collector != null) {
+      // collector expects currentDirectory to be the root of the dart
+      // package (i.e. contains lib/ and test/ sub-dirs).
+      fs.currentDirectory = testDirectory.parent;
+      if (!await
+          collector.collectCoverageData(argResults[_kOptionCoveragePath]))
+        throwToolExit('Failed to collect coverage data');
+    }
+
+    exit(result);
```
Comment (cbracken) [inline packages/flutter_tools/bin/fuchsia_tester.dart]: Optionally you can do the following above, and skip the exit call here.
```
exitCode = await runTests(...
```

File: packages/flutter_tools/bin/fuchsia_tester.dart [hunk 2]
```diff
@@ -70,16 +81,37 @@ Future<Null> run(List<String> args) async {
     if (!fs.isFileSync(shellPath)) {
       throwToolExit('Cannot find Flutter shell at $shellPath');
     }
-    loader.installHook(
-      shellPath: shellPath,
-    );
+    // Put the tester shell where runTests expects it.
+    final Link testerDestLink =
```
Comment (tvolkert) [inline packages/flutter_tools/bin/fuchsia_tester.dart]: Add a TODO to switch to a Fuchsia-specific `Artifacts` implementation.

**Review by tvolkert — APPROVED** [review]
LGTM

File: packages/flutter_tools/bin/fuchsia_tester.dart (hunk 2 above)
Comment (ricelid) [reply packages/flutter_tools/bin/fuchsia_tester.dart]: Done.

File: packages/flutter_tools/bin/fuchsia_tester.dart (hunk 1 above)
Comment (ricelid) [reply packages/flutter_tools/bin/fuchsia_tester.dart]: Done.

**Review by cbracken — APPROVED** [review]

**Comment by cbracken** [comment]
Re-LGTM. I'll land once the bots go green.

**Pull request merged**